### PR TITLE
Add a voiding mode for Super/Quantum tanks

### PR DESCRIPTION
**Pull request by MuXiu1997**
closed https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/7565

### CODE REVIEW

File: src/main/java/gregtech/common/tileentities/storage/GT_MetaTileEntity_DigitalTankBase.java [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+    }
+
+    @Override
+    public int getTankPressure() {
+        return 100;
+    }
+
+    @Override
+    public int getCapacity() {
+        return commonSizeCompute(mTier);
```
Comment (Glease) [inline src/main/java/gregtech/common/tileentities/storage/GT_MetaTileEntity_DigitalTankBase.java:269]: I'd say return Integer.MAX_VALUE if mVoidOverflow is on. There are smart logisitics tools that try to figure out how much to fill before calling #fill()

Comment (MuXiu1997) [reply src/main/java/gregtech/common/tileentities/storage/GT_MetaTileEntity_DigitalTankBase.java]: Awesome!

File: src/main/java/gregtech/common/tileentities/storage/GT_MetaTileEntity_DigitalTankBase.java [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+        }
+    }
+
+    @Override
+    public int fill(FluidStack aFluid, boolean doFill) {
+        if (aFluid == null || aFluid.getFluid().getID() <= 0 || aFluid.amount <= 0 || !canTankBeFilled() || !isFluidInputAllowed(aFluid))
+            return 0;
+
+        if (getFillableStack() == null || getFillableStack().getFluid().getID() <= 0) {
+            if (aFluid.amount <= getCapacity()) {
```
Comment (Glease) [inline src/main/java/gregtech/common/tileentities/storage/GT_MetaTileEntity_DigitalTankBase.java:218]: getCapacity is no longer reliable once we change perceived capacity when mVoidMode is on. We will probably need a getRealCapacity or something...

**Review by Glease — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/common/tileentities/storage/GT_MetaTileEntity_DigitalTankBase.java (hunk 2 above)
Comment (MuXiu1997) [reply src/main/java/gregtech/common/tileentities/storage/GT_MetaTileEntity_DigitalTankBase.java]: I was too careless
Fixed

**Review by Glease — APPROVED** [review]

**Pull request merged**